apps/heft/src/plugins/RunScriptPlugin.ts

```diff
@@ -1,134 +1,71 @@
-    heftEvent: HeftEvent,
-    runScriptEventActions: IHeftConfigurationRunScriptEventAction[],
-    logger: ScopedLogger,
-    heftSession: HeftSession,
+export default class RunScriptPlugin implements IHeftTaskPlugin<IRunScriptPluginOptions> {
+  public apply(
+    heftTaskSession: IHeftTaskSession,
     heftConfiguration: HeftConfiguration,
-    properties: ITestStageProperties
-  ): Promise<void> {
-    await this._runScriptsForHeftEventActions(
-      runScriptEventActions,
-      logger,
-      heftSession,
-      heftConfiguration,
-      properties
+    pluginOptions: IRunScriptPluginOptions
+  ): void {
+    heftTaskSession.hooks.run.tapPromise(
+      heftTaskSession.taskName,
+      async (runOptions: IHeftTaskRunHookOptions) => {
+        await this._runScriptAsync(heftTaskSession, heftConfiguration, pluginOptions, runOptions);
+      }
     );
   }
 
-  private async _runScriptsForHeftEventActions<TStageProperties>(
-    runScriptEventActions: IHeftConfigurationRunScriptEventAction[],
-    logger: ScopedLogger,
-    heftSession: HeftSession,
+  private async _runScriptAsync(
+    heftTaskSession: IHeftTaskSession,
     heftConfiguration: HeftConfiguration,
-    properties: TStageProperties
+    pluginOptions: IRunScriptPluginOptions,
+    runOptions: IHeftTaskRunHookOptions
   ): Promise<void> {
-    await Async.forEachAsync(
-      runScriptEventActions,
-      async (runScriptEventAction) => {
-        // The scriptPath property should be fully resolved since it is included in the resolution logic used by
-        // HeftConfiguration
-        const resolvedModulePath: string = runScriptEventAction.scriptPath;
-
-        // Use the HeftEvent.actionId field for the logger since this should identify the HeftEvent that the
-        // script is sourced from. This is also a bit more user-friendly and customizable than simply using
-        // the script name for the logger. We will also prefix the logger name with the plugin name to clarify
-        // that the output is coming from the RunScriptPlugin.
-        const scriptLogger: ScopedLogger = heftSession.requestScopedLogger(
-          `${logger.loggerName}:${runScriptEventAction.actionId}`
-        );
+    // The scriptPath property should be fully resolved since it is included in the resolution logic used by
+    // HeftConfiguration
+    const resolvedModulePath: string = pluginOptions.scriptPath;
 
-        const runScript: IRunScript<TStageProperties> = require(resolvedModulePath);
-        if (runScript.run && runScript.runAsync) {
-          throw new Error(
-            `The script at "${resolvedModulePath}" exports both a "run" and a "runAsync" function`
-          );
-        } else if (!runScript.run && !runScript.runAsync) {
-          throw new Error(
-            `The script at "${resolvedModulePath}" doesn\'t export a "run" or a "runAsync" function`
-          );
-        }
+    const runScript: IRunScript = await import(resolvedModulePath);
```

Comment:
I don't believe so? When `await import('...')` is transpiled, it transpiles to `await Promise.resolve().then(() => __importStar(require("...")));` since the tsconfig specifies `esModuleInterop: true`, which should still handle this scenario gracefully.

Reply:
And for reference, the original behaviour was to just do a `require(...)` call.